src/enableDisableRules.ts

```diff
@@ -63,10 +72,31 @@ export class EnableDisableRulesWalker extends SkippableTokenAwareRuleWalker {
                     if (!(ruleToAdd in this.enableDisableRuleMap)) {
                         this.enableDisableRuleMap[ruleToAdd] = [];
                     }
-                    this.enableDisableRuleMap[ruleToAdd].push({
-                        isEnabled: isEnabled,
-                        position: startingPosition,
-                    });
+                    if (isCurrentLine) {
+                        // start at the beginning of the current line
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: isEnabled,
+                            position: this.getStartOfLinePosition(node, startingPosition),
+                        });
+                        // end at the beginning of the next line
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: !isEnabled,
+                            position: scanner.getTextPos() + 1,
```

Comment:
The third param is line offset, which defaults to 0, the current line. For current line switches we want to start at the beginning of the current line, so line offset isn't necessary.